src/pages/RFOX/components/RewardsAndClaims/Claims.tsx

```diff
@@ -1,50 +1,84 @@
 
-  // @TODO: to be removed when wiring it up
-  const staticKey = '1'
+  const stakingAsset = useAppSelector(state => selectAssetById(state, stakingAssetId))
 
-  return (
-    <CardBody>
-      {headerComponent}
+  const stakingAssetAccountAddress = useMemo(
+    () => (stakingAssetAccountId ? fromAccountId(stakingAssetAccountId).account : undefined),
+    [stakingAssetAccountId],
+  )
 
-      <Box>
-        <ClaimRow
-          stakingAssetId={foxAssetId}
-          key={staticKey}
-          amountCryptoPrecision={'1500'}
-          status={ClaimStatus.CoolingDown}
-          setConfirmedQuote={setConfirmedQuote}
-          cooldownPeriodHuman={'in 1 day'}
-          index={1}
-          actionDescription={translate('RFOX.unstakeFrom', { assetSymbol: 'FOX' })}
-          displayClaimButton
-        />
+  const { data: unstakingRequestResponse, isLoading: isUnstakingRequestLoading } =
+    useGetUnstakingRequestQuery({ stakingAssetAccountAddress })
+
+  const claims = useMemo(() => {
+    if (!stakingAsset) return null
+
+    if (isUnstakingRequestLoading)
+      return new Array(2).fill(null).map(() => <Skeleton height={16} />)
+
+    return (unstakingRequestResponse ?? []).map((unstakingRequest, index) => {
+      const amountCryptoPrecision = fromBaseUnit(
+        unstakingRequest.unstakingBalance.toString(),
+        stakingAsset?.precision ?? 0,
+      )
+      const currentTimestampMs: number = Date.now()
+      const unstakingTimestampMs: number = Number(unstakingRequest.cooldownExpiry) * 1000
+      const isAvailable = currentTimestampMs >= unstakingTimestampMs
+      const cooldownDeltaMs = unstakingTimestampMs - currentTimestampMs
+      const cooldownPeriodHuman = dayjs(Date.now() + cooldownDeltaMs).fromNow()
+      const status = isAvailable ? ClaimStatus.Available : ClaimStatus.CoolingDown
+      return (
         <ClaimRow
-          stakingAssetId={foxAssetId}
-          key={staticKey}
-          amountCryptoPrecision={'1500'}
-          status={ClaimStatus.Available}
+          stakingAssetId={stakingAssetId}
+          key={unstakingRequest.cooldownExpiry.toString()}
+          amountCryptoPrecision={amountCryptoPrecision?.toString() ?? ''}
+          status={status}
           setConfirmedQuote={setConfirmedQuote}
-          cooldownPeriodHuman={'1 day ago'}
-          index={1}
-          actionDescription={translate('RFOX.unstakeFrom', { assetSymbol: 'FOX' })}
-          displayClaimButton
+          cooldownPeriodHuman={cooldownPeriodHuman}
+          index={index}
+          actionDescription={translate('RFOX.unstakeFrom', {
+            assetSymbol: stakingAsset.symbol,
+          })}
         />
-      </Box>
+      )
+    })
+  }, [
+    isUnstakingRequestLoading,
+    setConfirmedQuote,
+    stakingAsset,
+    stakingAssetId,
+    translate,
+    unstakingRequestResponse,
+  ])
+
+  if (!stakingAsset) return null
+
+  return (
+    <CardBody>
+      {headerComponent}
+      <Box>{claims}</Box>
```

Comment:
There are a few UI problems:

- When claims didn't load yet and there are no claims, we only have the title displayed, no skeleton or empty text
- Skeleton doesn't have any spacing, resulting in a big loading block which doesn't feel so beautiful
- If it loads and we have no claims, we see the title with nothing else, we should display a text "No claims available" as we previously seen this block already

Here is a small example of a diff:
https://gist.github.com/NeOMakinG/ae4a0c4742bc934d58de3622a5ce54e0

Reply:
Be careful the diff may require a bit more work: `if (isUnstakingRequestLoading || !unstakingRequestResponse)` might require to be a bit more reworked as we might end up in an infinite loop if the query didn't return any response but has been executed

Reply:
Tackled in https://github.com/shapeshift/web/commit/1b250fc48180834caa65c41546d9af94f5ce4f3d